templates/documents/shared/details_base.html

```diff
@@ -0,0 +1,28 @@
+<form
+    id="detail-form"
+    action="{{ edit_url }}"
+    method="POST"
+    class="m-2 grid grid-cols-4" x-data="{editing: false}">
+    {% csrf_token %}
+    {% block detail_content %}
+    {% endblock %}
+    <div class="py-3 col-span-1">
+        <div class="join">
+            <button type="button" class="btn btn-sm join-item pl-3" x-show="!editing" @click="editing = true">
+                <i class="fa-solid fa-pencil fa-xs"></i>
+            </button>
+            <button type="submit" class="btn btn-sm join-item" x-cloak x-show="editing">
+                <i class="fa-solid fa-check fa-xs"></i>
+            </button>
+            <a type="button" class="btn btn-sm join-item" x-show="!editing" href="{{ delete_url }}">
```

Comment:
This should be a form that does a POST or DELETE request. You can just wrap a button in a form and use the button to submit the form. Here's an example: https://github.com/dimagi/open-chat-studio/blob/2b45c04f0a8fbe163bbc37dfb22ab562a452016f/templates/account/components/api_keys.html#L44-L50

Reply:
See [this comment](https://github.com/dimagi/open-chat-studio/pull/1259#discussion_r2003210314)